src/snowflake/snowpark/session.py

```diff
@@ -1974,10 +1974,14 @@ def _table_exists(self, table_name: str):
         # note: object name could have dots, e.g, a table could be created via: create table "abc.abc" (id int)
         # currently validate_object_name does not allow it, but if in the future we want to support the case, we need to
         # update the implementation accordingly in this method
-        qualified_table_name = table_name.split(".")
```

Comment:
I think we allow `"db"."schema"."table"`, then will it break?

Reply:
I have updated the implementation, it should be able to handle all the special cases now